src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeExceptionHelpers.cs

```diff
@@ -192,23 +192,16 @@ public struct ExceptionInformationArray
         }
 
         private static ulong s_crashingThreadId;
-        private static CrashInfo? s_crashInfo;
-        private static object s_crashLockObject = new object();
+        private static CrashInfo s_crashInfo;
+        private static int s_crashInfoPresent;
 
         internal static CrashInfo SerializeCrashInfo(RhFailFastReason reason, string? message, Exception? exception)
         {
-            if (s_crashInfo != null)
+            int previousState = Interlocked.CompareExchange(ref s_crashInfoPresent, 1, 0);
+            if (previousState == 0)
             {
-                // If we have already reported a crash, return the existing crash info.
-                return s_crashInfo.Value;
-            }
+                s_crashInfoPresent = 1;
```

Comment:
Isn't this assignment redundant? `s_crashInfoPresent` should always be set to `1` already here, no? 🤔